### ISSUE TITLE
Test-Time Training阶段计算CIM损失，超出范围IndexError: tuple index out of range

### ISSUE THREAD

**Issue by luming1314** [ziwang1121/HTT#1]
作者，您好！我们在RGBNT201数据集训练test_time_train.py时，CIM损失会报IndexError: tuple index out of range。
我们debug发现，label_num显示是12，而feat1的最大范围是8，意味着超出了元组的范围。请问这个是什么原因？
![image](https://github.com/user-attachments/assets/00aa724b-dc7c-48c5-adb1-c16decf07bac)

**Comment by muhammadumair894** [comment]
@luming1314 Hello. Did you solve your query?

**Comment by luming1314** [comment]
Thank you. I have solved it

**Comment by ziwang1121** [comment]
> [@luming1314](https://github.com/luming1314) Hello. Did you solve your query?

Hello, thank you for your interest in our work.
As far as I recall, we set the batch size to 2 during the test-time training phase—you may try this setting.

Here’s why:

In ReID tasks, the sampling strategy during testing can differ from that during training when using triplet loss. For example, with a batch size of 16, if we use a sampler for cross-entropy (CE) loss only, the sampled identities are completely random. However, a triplet-loss-based sampler enforces sampling of 4 images from 4 identities (i.e., 16 images in total).

If random sampling is used and the batch size is larger than 2, the CIM loss might not work properly.
For instance, suppose the sampled ID list is: [1, 3, 4, 5, 6]. In this case, the CIM loss runs normally since label_number = 5, and the features can be evenly split into 5 groups.
However, when the sampled list includes repeated IDs, such as [1, 3, 3, 4, 4], then the CIM loss might fail. Here, label_number = 3, but the features would still be split into 5/3 = 2 parts (automatically rounded). If identity 1 only has one sample (i.e., only feat[0]), then feat[1] doesn't exist, which results in the out-of-bounds error you mentioned.

I believe this is the cause of the issue. You may try setting the batch size to 2 and see if that resolves it.
Feel free to reach out if you have further questions.

Best regards,
Zi Wang

**Comment by ziwang1121** [comment]
> [@luming1314](https://github.com/luming1314) Hello. Did you solve your query?

We have now shared the code (see the .zip file), which we have verified to be fully reproducible. For setup details (e.g., environment and commands), please refer to the instructions on the GitHub repository.

https://drive.google.com/file/d/199jCO4F9kFvDn5BC59YnGaJgq2AgxBH-/view?usp=drive_link

We will also update the repository with further improvements shortly.
